### ISSUE TITLE
Would be great to have global filter on class level annotation [DATAGRAPH-1348]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-data-neo4j#1910]
**[Gerrit Meier](https://jira.spring.io/secure/ViewProfile.jspa?name=gerrit.meier)** opened **[DATAGRAPH-1348](https://jira.spring.io/browse/DATAGRAPH-1348?redirect=false)** and commented

It would be great to have an annotation on class to filter queries automatically with any requests.
For example, during the soft deletion developers used to set deleted date or some other deleted flags to mark the entity as deleted. So in your case, it will be the Node.

```
@Where(clause = "deleted is null")
```

The following code snippet shows a `@where` annotation that excludes a record if its state is DELETED. This is used in Hibernate tech stack.
So what it does? When someone calls findById(id) and the node is found but the "deleted" date is set there it counts it as deleted because of `@where` and does not return the Node.
This helps when in your whole project everything is soft deletable you should write in any query the part which excludes the Nodes which have deleted property with date already set.
findById -----> findByIdAndDeletedIsNull . And any other queries should have this suffix "**andDeletedIsNull"
The annotation could be `@filter` and in that case, it would work as permitter.
it could be smth like this: ```
`@Filter`(where = "deleted is null")

```and we could put it on class level.

The example concerns soft deletion but I guess it could serve globally for everything
```

---

**Affects:** 6.0 M2 (2020.0.0)

**Reference URL:** https://github.com/neo4j/sdn-rx/issues/212

**Comment by michael-simons** [comment]
Now that SDN 6 is live for a couple of months it is time to go back to this feature request @aaramg. Thanks again for your interest. 

Given now the experience with SDN 6 this is not something we can or want to implement. The reason is rather simple: SDN 6 assumes the fetched object graph to be "the truth", that is, when stored back, it will replace the previous set.

While the suggested `@Filter` annotation would not prohibit that behavior for the top level nodes being fetched, it would do so for related nodes: When I would define such a filter, I would expect that it is applied when I fetch the node annotated itself or via a relationship. When it's applied on a related node however, the related node would not appear in the graph of fetched objects and would be deleted when the graph of fetched object is persisted.

Basically, the same arguments why we are not adding a fetch depth apply here.

If you need to shape the result set in a different form, please use projections, custom queries or combinations thereof.
